front/components/assistant_builder/ActionsScreen.tsx

```diff
@@ -122,12 +132,65 @@ export default function ActionsScreen({
   setAction: (action: AssistantBuilderSetActionType) => void;
   pendingAction: AssistantBuilderPendingAction;
 }) {
+  const { vaults } = useContext(AssistantBuilderContext);
+
   const configurableActions = builderState.actions.filter(
     (a) => !(CAPABILITIES_ACTION_CATEGORIES as string[]).includes(a.type)
   );
 
   const isLegacyConfig = isLegacyAssistantBuilderConfiguration(builderState);
 
+  const vaultIdToActions = useMemo(() => {
```

Comment:
IIRC, @philipperolet introduced a `getDataSourceViewIdsFromActions` method, can we use it to ensure there is only one place where we need to perform this operation?

Reply:
Just saw it after merging.

Reply:
Looked..

Not super practical for that use case (it takes form body, returns datasourceviewids).
I'll keep as it.